packages/lesswrong/server/callbacks.ts

```diff
@@ -44,7 +44,9 @@ getCollectionHooks("Users").editAsync.add(async function userEditNullifyVotesCal
 });
 
 getCollectionHooks("Users").editAsync.add(async function userEditChangeDisplayNameCallbacksAsync(user: DbUser, oldUser: DbUser) {
-  if (user.displayName !== oldUser.displayName) {
+  // on the EA Forum, we assign new users a randomized "new_user_#" which we prompt them to change right away,
+  // and we don't want this action to count toward their one username change
+  if (user.displayName !== oldUser.displayName && !oldUser.displayName.startsWith('new_user_')) {
```

Comment:
This is a strict improvement, but doesn't catch the related case, where they came in with a username from their OAuth provider. It would be ideal if the new user signup page didn't add to previousDisplayName at all.

Reply:
Oh, how can a new user "come in with a username"? In that case, wouldn't they see that username in the form? Although I guess we shouldn't count changing that as their one username change?

However, I'm not sure how to exclude all submissions from this one form from counting as the one username change.

Reply:
Yes, they would see it in the form. It's because Auth0 might know their name (like from EA Global applications). You could check whether the old user has `usernameUnset`. I'm pretty sure that would work.

Reply:
Quite confident that would work.